test/ngResource/resourceSpec.js

```diff
@@ -978,6 +979,194 @@ describe("basic usage", function() {
       });
     });
 
+
+    describe('requestInterceptor', function() {
+      var rejectReason = {'lol':'cat'};
+      var testError = new Error();
+
+      it('should allow per action request interceptor that gets full configuration', function() {
+        CreditCard = $resource('/CreditCard', {}, {
+          query: {
+            method: 'get',
+            isArray: true,
+            interceptor: {
+              request: function(httpConfig) {
+                callback(httpConfig);
+                return httpConfig;
+              }
+            }
+          }
+        });
+
+        $httpBackend.expect('GET', '/CreditCard').respond([{id: 1}]);
+
+        CreditCard.query();
+
+        $httpBackend.flush();
+        expect(callback).toHaveBeenCalledOnce();
+
+        var config = callback.mostRecentCall.args[0];
+        expect(config.method).toBe('get');
+        expect(config.url).toBe('/CreditCard');
+      });
+
+      it('should call $http with the value returned from requestInterceptor', function() {
+        CreditCard = $resource('/CreditCard', {}, {
+          query: {
+            method: 'get',
+            isArray: true,
+            interceptor: {
+              request: function(httpConfig) {
+                httpConfig.url = '/DebitCard';
+                return $q.resolve(httpConfig);
+              }
+            }
+          }
+        });
+
+        $httpBackend.expect('GET', '/DebitCard').respond([{id: 1}]);
+
+        CreditCard.query();
+
+        $httpBackend.flush();
+      });
+
+      it('should abort the operation if the requestInterceptor rejects the operation', function() {
+        var rejectResponse;
+
+        CreditCard = $resource('/CreditCard', {}, {
+          query: {
+            method: 'get',
+            isArray: true,
+            interceptor: {
+              request: function() {
+                return $q.reject(rejectReason);
+              }
+            }
+          }
+        });
+
+        // This doesn't have a corresponding $httpBackend.expect call, so verify
+        var card = CreditCard.query();
+        card.$promise.catch(function(rejection) {
+          rejectResponse = rejection;
+        });
+
+        // Make sure all promises resolve.
+        $rootScope.$apply();
+
+        // Ensure the resource promise was rejected
+        expect(card.$resolved).toBeTruthy();
+        expect(card.$promise.$$state.status).toEqual(2);
```

Comment:
This is private API surface. We don't need to verify that. (It might change anytime anyway.)